test/jdk/java/net/URLConnection/RequestProperties.java

```diff
@@ -79,4 +81,29 @@ private static boolean hasFtp() {
             return false;
         }
     }
+
+    /**
+     * Test that {@link URLConnection#setRequestProperty(String, String)} throws
+     * an {@link IllegalStateException} when already connected
+     *
+     * @throws Exception
+     */
+    private static void testSetRequestPropIllegalStateException() throws Exception {
+        final URL url = new URL("file://" + System.getProperty("java.io.tmpdir"));
+        final java.net.URLConnection conn = url.openConnection();
```

Comment:
The URL creation isn't technically right as you can't reliably use a file path string as a URL path. Something like the following would be more correct:
`URL url = Path.of(System.getProperty("java.io.tmpdir")).toUri().toURL();`

Also minor nit but importing java.net.URL and not URLConnection is a bit strange.

Reply:
> The URL creation isn't technically right as you can't reliably use a file path string as a URL path. Something like the following would be more correct:
URL url = Path.of(System.getProperty("java.io.tmpdir")).toUri().toURL();

I've now updated the PR to fix this.

> Also minor nit but importing java.net.URL and not URLConnection is a bit strange.

Sorry, that wasn't intentional. Not sure how I managed to do that. I've included this change in the updated PR.